spec/enumerable_spec.rb

```diff
@@ -0,0 +1,160 @@
+require('./lib/enumerable.rb')
+include Enumerable
+
+class MockObject
+  attr_reader :variable, :other_variable
+  def initialize variable = 'hello', other_variable = 'world'
+    @variable = variable
+    @other_variable = other_variable
+  end
+end
+
+RSpec.describe Enumerable do
+  describe "#my_each" do
+    it "should access all element in given array" do
+      counter = 0
+      array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      array.my_each do |e|
+        expect(e).to eql(array[counter])
+        counter += 1
+      end
+      expect(counter).to eql(array.length)
+    end
+
+    it "should work with object data" do
+      object_array = [MockObject.new, MockObject.new, MockObject.new]
+      object_array.my_each do |e|
+        expect(e.variable).to eql('hello')
+        expect(e.other_variable).to eql('world')
+      end
+    end
+
+  end
+  describe "#my_each_with_index" do
+    it "should return an element by index" do
+      array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      expect(array.my_each_with_index(0)).to eql(1)
```

Comment:
This is not how each_with_index works. Take a look [here](https://ruby-doc.org/core-2.6/Enumerable.html#method-i-each_with_index)

Reply:
We didn't get the concept behind each_with_index. It was changed and ready to review by you!